docs/gallery/general/plot_configurator.py

```diff
@@ -0,0 +1,108 @@
+they want to be validated. After creating the configuration file, users will need to load the
+configuration with the :py:func:`~pynwb.load_type_config` method.
+With the configuration loaded, every instance of the neurodata
+types defined in the configuration file will have the respective fields wrapped with a
+:py:class:`~hdmf.term_set.TermSetWrapper`.
+This automatic wrapping is what provides the term validation for the the field value.
+If a user wants to have greater control on which instances have validated fields, the user cannot use the
+configurator, bur rather proceed with manually wrapping with a
+:py:class:`~hdmf.term_set.TermSetWrapper`.
+
+To unload a configuration, simply call :py:func:`~pynwb.unload_type_config`.
+We also provide a helper method to see the configuration that has been loaded:
+:py:func:`~pynwb.get_loaded_type_config`
+
+
+How to make a Configuration File
+--------------------------------
+Before taking advantage of the all the wonders that comes with using a configuration file,
+the user needs to create one following some simple guidelines. To follow along with an example,
+please refer to ``nwb_config.yaml`` under ``src/config``.
+The configuration files is built on the foundation of the YAML syntax. The
+user will construct a series of nested dictionaries to encompass all the necessary information.
+
+1. The user needs to define all the relevant namespaces. Recall that each neurodata type exists within
+   a namespace, whether that is the core namespace in PyNWB or a namespace in an extension. As namespaces grow,
+   we also require a version to be recorded in the configuration file to ensure proper functionality.
+2. Within a namespace dictionary, the user will have a list of data types the want to use.
+3. Each data type will have a list of fields associated with a :py:class:`~hdmf.term_set.TermSet`.
+   The user can use the same or unique TermSet instances for each field.
+"""
+try:
+    import linkml_runtime  # noqa: F401
+except ImportError as e:
+    raise ImportError("Please install linkml-runtime to run this example: pip install linkml-runtime") from e
+
+from dateutil import tz
+from datetime import datetime
+from uuid import uuid4
+import os
+
+from pynwb import NWBFile, get_loaded_type_config, load_type_config, unload_type_config
+from pynwb.file import Subject
+
+# How to use a Configuration file
+# -------------------------------
+# As mentioned prior, the first step after creating a configuration file is
+# to load the file.
+# It is important to remember that with the configuration loaded, the fields
+# are wrapped automatically, meaning the user should proceed with creating
+# the instances normally, i.e., without wrapping directly. In this example,
+# we load the a configuration file that associates a
+# :py:class:`~hdmf.term_set.TermSet` for the ``species`` field in
+# :py:class:`~pynwb.file.Subject`.
+
+dir_path = os.path.dirname(os.path.abspath(__file__))
```

Comment:
```suggestion
dir_path = os.path.dirname(os.path.abspath("plot_configurator.py"))
```
I think `__file__` does not exist when running under certain types of scripts or environments. I think this should work in its place.

Reply:
We did this in the icephys tutorial:
```python
try:
	tutorial_path = os.path.abspath(__file__)  # when running as a .py
except NameError:
	tutorial_path = os.path.abspath("__file__")  # when running as a script or notebook
```

Reply:
What's weird is that __file__ is what we have in hdmf and it works

Reply:
That is weird. I don't know. If you want to track it down, you could look into differences in sphinx versions or the link checker call.